handbook/engineering/search/index.md

```diff
@@ -2,9 +2,93 @@
 
 We own the end-to-end Sourcegraph search experience.
 
+## Vision
+
+Sourcegraph search is:
+
+- **Fast:** Performance of showing results is fast and I can get to what I’m looking for quickly.
+- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write.
+The answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
+- **Easy to use:** Users can figure out how to find what they are looking for, the syntax is clear, new users understand what options they have for searching.
+- **Sharable:** Sourcegraph search creates network effects that provide compounding value with scale. 
+
 ## Goals
 
-Our current goals are documented in the [search team's current tracking issue](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+_Updated 2020-08-27_
+
+## Long term (~1 year)
+
+Sourcegraph search is far and away the best code search solution, search results are fast, and I can find what I’m looking for quickly.
+
+## Medium term (3-6 months)
+
+In the next 6 months we want to be halfway to the long term goal. To achieve this, we need to accomplish the following outcomes:
+
+- Getting to search results faster through a better search experience.
+- What do we need to do to get to the speed of grep.app?
```

Comment:
I don't know that grep.app is faster. It is perhaps perceptually faster because I think it only searches a couple of repos by default. The goal here might be to rather deliver a faster search experience, in which case, maybe we should state this in terms of desired time-to-result for users, and what scope of repos we want to be included in such an experience. Overall I'm not a big fan of defining goals by naming alternative solutions, could we extract the attributes that those solutions have and define our goals in terms of them? what is it about the speed of grep.app that appeals?

Reply:
Extracted **fast-loading** and **responsive** as key attributes that make grep.app so appealing.